src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -2238,9 +2266,15 @@ private void processJavaCompilation(File outputDirectory, File testOutputDirecto
         }
     }
  
-    private void checkStopDevMode() throws PluginScenarioException {
+    private void checkStopDevMode(boolean skipOnRestart) throws PluginScenarioException {
         // stop dev mode if the server has been stopped by another process
         if (serverThread == null || serverThread.getState().equals(Thread.State.TERMINATED)) {
+            // server is restarting if devStop was set to true and we have not called the shutdown hook
+            boolean restarting = devStop.get() && !calledShutdownHook.get();
```

Comment:
I noticed that just `devStop` is used here, but we use `this.devStop` below. Is there a reason to use `this`? If not, which is the better practice? We should probably be consistent.

Reply:
Good catch, `this` should not be necessary.